### PR TITLE
VMware: refactor configure disk logic

### PR DESCRIPTION
**Pull request by Akasurde**
##### SUMMARY
This fix corrects logic related to disk parameters.

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

##### ISSUE TYPE
 - Bugfix Pull Request

##### COMPONENT NAME
lib/ansible/modules/cloud/vmware/vmware_guest.py

##### ANSIBLE VERSION

```
2.6devel
```

### CODE REVIEW

**Comment by Akasurde** [comment]
@pdellaert @tchernomax Could you please review this ?

File: lib/ansible/modules/cloud/vmware/vmware_guest.py [hunk 1]
```diff
@@ -1323,29 +1323,47 @@ def get_vm_scsi_controller(self, vm_obj):
...
+                    else:
+                        # We found int value in string, let's typecast it
+                        expected = int(expected)
+                except (TypeError, ValueError, NameError):
+                    disk_size_parse_failed = True
+
+                if disk_size_parse_failed or not expected or not unit:
+                    # Common failure
+                    self.module.fail_json(msg="Failed to parse disk size please review value"
+                                              " provided using documentation.")
```
Comment (tchernomax) [inline lib/ansible/modules/cloud/vmware/vmware_guest.py]: I propose:
```
if [x for x in expected_disk_spec.keys() if x.startswith('size_') or x == 'size']:
    # size, size_tb, size_gb, size_mb, size_kb
    if 'size' in expected_disk_spec:
        size_regex = re.compile(r'(\d+(?:\.\d+)?)([tgmkTGMK][bB])')
        disk_size_m = size_regex.match(expected_disk_spec['size'])
        try:
            if disk_size_m:
                expected = disk_size_m.group(1)
                unit = disk_size_m.group(2)
            else:
	    	raise ValueError

            if re.match(r'\d+\.\d+', expected):
                # We found float value in string, let's typecast it
                expected = float(expected)
            else:
                # We found int value in string, let's typecast it
                expected = int(expected)

            if not expected or not unit:
	    	raise ValueError
		
        except (TypeError, ValueError, NameError):
            # Common failure
            self.module.fail_json(msg="Failed to parse disk size please review value"
                                      " provided using documentation.")
```

No need to add the following:
```
+        disk_size_parse_failed = False
+        expected = None
+        unit = None
```

@Akasurde What do you think ?

**Review by tchernomax — CHANGES_REQUESTED** [review]

File: lib/ansible/modules/cloud/vmware/vmware_guest.py (hunk 1 above)
Comment (Akasurde) [reply lib/ansible/modules/cloud/vmware/vmware_guest.py]: Done.

**Comment by Akasurde** [comment]
@tchernomax Done.

**Review by tchernomax — APPROVED** [review]
LGTM

**Comment by Akasurde** [comment]
@tchernomax Thanks for your valuable reviews and time.

**Pull request merged**